### PR TITLE
User/susohoni/wmc win32 sample

### PR DESCRIPTION
**Pull request by Sudhaushu**
Refactor code, added mock preview to drain samples from pipeline and allow 3A to converge

### CODE REVIEW

File: Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp [hunk 1]
```diff
@@ -31,115 +31,247 @@ using namespace Windows::Storage;
...
+    //
+    IVector<MediaFrameSourceInfo> GetFilteredSourceGroupList()
     {
-        auto sourceInfos = sourceGroupIter.Current().SourceInfos();
-        auto sourceInfoIter = sourceInfos.First();
-        // Iterate through sources and filter-out the IR, depth and other sources which we cannot consume in this app
-        while (sourceInfoIter.HasCurrent())
+        auto filteredSourceInfos = single_threaded_vector<MediaFrameSourceInfo>();
+        m_SourceGroups = MediaFrameSourceGroup::FindAllAsync().get();
+        auto sourceGroupIter = m_SourceGroups.First();
```
Comment (LPBourret) [inline Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp]: catch case where there are no source group returned

File: Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp [hunk 2]
```diff
@@ -31,115 +31,247 @@ using namespace Windows::Storage;
...
+            streamTypelookup = (streamTypelookup == MediaStreamType::VideoPreview) ? MediaStreamType::VideoRecord : MediaStreamType::Photo;
         }
 
-        std::wcout << idx << L":" << currSource.SourceGroup().DisplayName().c_str() <<"-->" << currSource.DeviceInformation().Name().c_str() << panelLocation.c_str() << ":" << streamTypes[currSource.MediaStreamType()].c_str() << std::endl;
-        source.MoveNext();
+        m_PreviewFrameReader = mediaCapture.CreateFrameReaderAsync(previewframeSource).get();
+        m_PreviewFrameReader.AcquisitionMode(MediaFrameReaderAcquisitionMode::Realtime);
+        m_PreviewFrameReader.StartAsync().get();
+        std::cout << "\nDraining samples for one second from the pipeline to enable 3A convergence\n";
+        Sleep(1000);
```
Comment (LPBourret) [inline Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp]: >Sleep(1000) [](start = 8, length = 11)

suggestion: you can substitue with "co_await 1s;"for a C++/WinRT way of awaiting passively

File: Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp [hunk 3]
```diff
@@ -31,115 +31,247 @@ using namespace Windows::Storage;
...
+        }
+
+        StartMockPreview(mediaCapture);
+
+        return mediaCapture;
     }
-    int selection;
-    do
+
+    void DeInitCamera(MediaCapture &mediaCapture)
```
Comment (LPBourret) [inline Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp]: add header to all your methods

**Comment by LPBourret** [comment]
maybe specify on your README.MD that there is a MockPreview instantiated to allow 3A to converge and why it's important

File: Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp (hunk 1 above)
Comment (Sudhaushu) [reply Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp]: In this case, this function simply returns a empty vector. Further , this case is handled at line 215. 

---
In reply to: [286775667](https://github.com/microsoft/Windows-Camera/pull/10#discussion_r286775667) [](ancestors = 286775667)

File: Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp (hunk 2 above)
Comment (Sudhaushu) [reply Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp]: but the idea here is to block the further execution and not take photos for 1sec. With co_await eventually up the stack I need to put this in a Task and as I understand co_await will only keep this Task call stack active for 1s and proceed execution of the rest of the routines, unless I wait, which is the same effect. The rest of the sample does not use co_routines directly to keep things simple.

---
In reply to: [286776586](https://github.com/microsoft/Windows-Camera/pull/10#discussion_r286776586) [](ancestors = 286776586)

**Pull request merged**